editor/editor_paths.cpp

```diff
@@ -118,11 +118,7 @@ EditorPaths::EditorPaths() {
 	// Self-contained mode if a `._sc_` or `_sc_` file is present in executable dir.
 	String exe_path = OS::get_singleton()->get_executable_path().get_base_dir();
 
-	// On macOS, look outside .app bundle, since .app bundle is read-only.
-	if (OS::get_singleton()->has_feature("macos") && exe_path.ends_with("MacOS") && exe_path.path_join("..").simplify_path().ends_with("Contents")) {
-		exe_path = exe_path.path_join("../../..").simplify_path();
-	}
-	{
+	if (!OS::get_singleton()->has_feature("macos")) {
```

Comment:
Since you say that from the terminal it does work, should we actually keep it but add a warning that it will not work from Finder if we find the `_sc_` file at the expected location?

Reply:
> Since you say that from the terminal it does work,

It will work only for a Project Manager (or if you open project with a `--path` argument), as soon as project is opened it will not, so it's not really working.

Reply:
Well, it does work under some conditions, so removing our support completely doesn't help the users who are currently affected, but also hurts the user who could benefit from launching directly via CLI. I think detecting the self-contained mode when launched from the terminal and printing a warning is a good middle ground here. And of course the change to the docs.

Reply:
I guess why not. Also re-added 3.x style behavior support, since it will work in some cases as well. But it still should be documented as unsupported and should not be used for official Steam releases.